fenix/app/src/main/java/org/mozilla/fenix/settings/quicksettings/protections/cookiebanners/CookieBannerHandlingUtils.kt

```diff
@@ -0,0 +1,59 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.quicksettings.protections.cookiebanners
+
+import android.content.Context
+import mozilla.components.browser.state.state.SessionState
+import mozilla.components.concept.engine.cookiehandling.CookieBannersStorage
+import org.mozilla.fenix.ext.settings
+import org.mozilla.fenix.trackingprotection.CookieBannerMode
+
+/**
+ * Util class for cookie banner handling reducer.
+ */
+object CookieBannerHandlingUtils {
+
+    /**
+     * Get the current status of cookie banner handling mode.
+     */
+    suspend fun getCookieBannerMode(
+        context: Context,
+        tab: SessionState,
+        cookieBannersStorage: CookieBannersStorage,
+        cookieBannerModeResult: (cookieBannerMode: CookieBannerMode) -> Any,
+    ) {
+        if (context.settings().shouldUseCookieBanner) {
+            if (cookieBannersStorage.hasException(
```

Comment:
I think this should run on an `IO` dispatcher :)

Reply:
Done